Makefile

```diff
@@ -307,30 +306,39 @@ functional-with-fault-injection-test: clean-test-results
 
 test: unit-test integration-test functional-test functional-with-fault-injection-test
 
-##### Coverage #####
+##### Coverage & Reporting #####
```

Comment:
I decided to keep things simple and merge "coverage" and "test reporting" into one. I don't think there's value in running them separately. And if devs run this locally, the extra test report step causes no harm.